docs/python/tutorial-azure-functions.md

```diff
@@ -0,0 +1,348 @@
+- The body of `main` then does whatever it wants to process the request and generate a response. In this case, the function looks for a `name` parameter in the URL. Failing that, it checks if the request body contains JSON (using the `func.HttpRequest.get_json` function) and that the JSON contains a `name` value (using the `get` method of the JSON object returned by `get_json`).
+- If a name is found, the function returns the string "Hello" with the name appended; otherwise it returns an error message.
+
+## Test and debug the function locally
+
+1. When you create the Functions project, the VS Code extension also creates a launch configuration in `.vscode/launch.json` that contains a single configuration named **Attach to Python Functions**. This configuration means you can just press F5 or use the Debug explorer to start the project:
+
+    ![Debug explorer showing the Functions launch configuration](images/functions/launch-configuration.png)
+
+1. When you start the debugger, a terminal opens showing output from Azure Functions, including a summary of the available endpoints:
+
+    ```
+    Hosting environment: Production
+    Content root path: d:\Examples\Python\AzureFunctions
+    Now listening on: http://0.0.0.0:7071
+    Application started. Press Ctrl+C to shut down.
+
+    Http Functions:
+
+            HttpExample: [GET,POST] http://localhost:7071/api/HttpExample
+    ```
+
+1. Ctrl+click the URL in the output window to open a browser to that address, or start a browser and paste in the same URL. In either case, you can see that the endpoint is `api/<function_name>`, in this case `api/HttpExample`. However, because that URL doesn't include a name parameter, the browser window should just show, "Please pass a name on the query string or in the request body" as appropriate for that path in the code.
+
+1. Now try the URL, `http://localhost:7071/api/HttpExample?name=VS%20Code` and you should see the message, "Hello VS Code!", demonstrating that you've run that code path.
+
+1. To pass the name value in a JSON request body, you can use a tool like curl with the JSON inline:
+
+    ```bash
+    # Mac OS/Linux
+    curl --header "Content-Type: application/json" --request POST \
+        --data {"name":"VS Code"} http://localhost:7071/api/HttpExample
+    ```
+
+    ```ps
+    # Windows (escaping on the quotes is necessary)
+    curl --header "Content-Type: application/json" --request POST \
+        --data {"""name""":"""VS Code"""} http://localhost:7071/api/HttpExample
+    ```
+
+    Alternately, create a file like `data.json` that contains `{"name":"VS Code"}` and use the command `curl --header "Content-Type: application/json" --request POST --data @data.json http://localhost:7071/api/HttpExample`.
```

Comment:
Remove? Not sure if this is beneficial.

Reply:
I wanted to include that detail to complete the thought of using CURL, so if you create your own function that expects more complex JSON, you don't have to go figure out how to use a file instead of the inline data. It's minimal impact with a potentially large benefit for a decent number of people.